packages/request-manager/src/torIdentities.ts

```diff
@@ -0,0 +1,91 @@
+import http from 'http';
+import https from 'https';
+import { SocksProxyAgent } from 'socks-proxy-agent';
+import { EventEmitter } from 'events';
+import { RequestInit } from 'node-fetch';
+
+interface TorConnectionOptions {
+    host: string;
+    port: number;
+}
+
+interface TorRequestOptions {
+    identity?: string;
+}
+
+export class TorIdentities extends EventEmitter {
+    options: TorConnectionOptions;
+    identities: { [key: string]: SocksProxyAgent };
+
+    constructor(options: TorConnectionOptions) {
+        super();
+        this.options = options;
+        this.identities = {
+            default: new SocksProxyAgent({
+                hostname: this.options.host,
+                port: this.options.port,
+                userId: 'default',
+                password: 'default',
+            }),
+        };
+    }
+
+    getIdentity(identity: string): SocksProxyAgent {
+        if (!this.identities[identity]) {
+            this.identities[identity] = new SocksProxyAgent({
+                hostname: this.options.host,
+                port: this.options.port,
+                userId: identity,
+                password: identity,
+            });
+        }
+        return this.identities[identity];
+    }
+
+    request(url: string, init: RequestInit, torRequestOptions?: TorRequestOptions) {
+        const requestUrl = new URL(url);
+        const agent = this.getIdentity(torRequestOptions?.identity || 'default');
+        const body = init?.body;
+        const options = {
+            method: init?.method || 'GET',
+            hostname: requestUrl.hostname,
+            port: requestUrl.port,
+            path: requestUrl.pathname + requestUrl.search,
+            headers: (init?.headers || {}) as http.IncomingHttpHeaders,
+            agent,
+            body,
+        };
+
+        return new Promise((resolve, reject) => {
```

Comment:
```suggestion
        return new Promise<string>((resolve, reject) => {
```
otherwise response type is `unknown`

Reply:
Changed https://github.com/trezor/trezor-suite/pull/6017/commits/9d574d83e0e1889a8afa6b20f42fcac7de1a21be#diff-e60ab2cf4bb18301168e666f64b7b593b417a543c5ec63f47a7e26cd6ca9175aR57